### PR TITLE
Used subTest() in forms.URLField() tests.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

File: tests/forms_tests/field_tests/test_urlfield.py [hunk 1]
```diff
@@ -7,150 +7,117 @@
...
+                with self.assertRaisesMessage(ValidationError, msg):
+                    f.clean(value)
+
+    def test_urlfield_clean_required(self):
+        f = URLField()
+        msg = "'This field is required.'"
+        with self.assertRaisesMessage(ValidationError, msg):
+            f.clean(None)
+        with self.assertRaisesMessage(ValidationError, msg):
+            f.clean('')
```
Comment (hramezani) [inline tests/forms_tests/field_tests/test_urlfield.py:115]: Can't we use `subTest` in this test?

File: tests/forms_tests/field_tests/test_urlfield.py [hunk 2]
```diff
@@ -7,150 +7,117 @@
...
-        self.assertWidgetRendersTo(f, '<input id="id_f" type="url" name="f" maxlength="20" minlength="15" required>')
-        with self.assertRaisesMessage(ValidationError, "'Ensure this value has at least 15 characters (it has 12).'"):
+        self.assertWidgetRendersTo(
+            f,
+            '<input id="id_f" type="url" name="f" maxlength="20" minlength="15" required>',
+        )
+        msg = "'Ensure this value has at least 15 characters (it has 12).'"
+        with self.assertRaisesMessage(ValidationError, msg):
             f.clean('http://f.com')
+        msg = "'Ensure this value has at most 20 characters (it has 37).'"
```
Comment (hramezani) [inline tests/forms_tests/field_tests/test_urlfield.py:23]: I would move it right before the `with self.assertRaisesMessage(ValidationError, msg):`

File: tests/forms_tests/field_tests/test_urlfield.py (hunk 1 above)
Comment (felixxm) [reply tests/forms_tests/field_tests/test_urlfield.py:115]: It seems like overkill just for two cases, and it's unlikely that we will have more.

File: tests/forms_tests/field_tests/test_urlfield.py (hunk 2 above)
Comment (felixxm) [reply tests/forms_tests/field_tests/test_urlfield.py:25]: Changed :+1:

**Comment by felixxm** [comment]
@hramezani Thanks for the review :+1:

**Review by hramezani — APPROVED** [review]

**Comment by lokeshchoudhury** [comment]
Please note,

Application test: rev: egg: test main break case is : "250382"

Apply pass the Password of the ransom ware that has been locking the premiering the Host: 8.8.0.5.132.192.9: please update and acknowledge.

Regards,
Lokesh C.


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Pull request merged**